src/dashboard/cd-dashboard-events.vue

```diff
@@ -1,32 +1,127 @@
 <template>
   <div class="row">
     <div class="cd-dashboard-events">
-      <h1 class="cd-dashboard-events__header">Upcoming Events</h1>
+      <div class="cd-dashboard-events__content">
+        <h1 class="cd-dashboard-events__header">{{ $t('Hey {name}, here\'s what\'s most important...', { name: loggedInUser.firstName }) }}</h1>
+        <div class="cd-dashboard-events__list">
+          <upcoming-event v-for="event in events" :key="event.id" :event="event" :dojo="dojos[event.dojoId]"></upcoming-event>
+        </div>
+        <div class="cd-dashboard-events__cta">
+          <router-link class="cd-dashboard-events__cta-link" :to="{ name: 'MyTickets' }">{{ $t('Your Events') }}</router-link>
+        </div>
+      </div>
     </div>
   </div>
 </template>
 
 <script>
+  import { mapGetters } from 'vuex';
+  import moment from 'moment';
+  import DojosService from '@/dojos/service';
+  import EventService from '@/events/service';
+  import EventUtils from '@/events/util';
+  import UpcomingEvent from './events/cd-dashboard-upcoming-event';
+
   export default {
     name: 'cd-dashboard-events',
+    components: {
+      UpcomingEvent,
+    },
+    data() {
+      return {
+        events: [{}, {}],
+        usersDojos: [],
+        dojos: {},
+        loaded: false,
+      };
+    },
+    computed: {
+      ...mapGetters(['loggedInUser']),
+    },
+    methods: {
+      async loadEvents() {
+        const query = { status: 'published' };
+        query.afterDate = moment().unix();
+        query.utcOffset = moment().utcOffset();
+
+        const events = await Promise.all(this.usersDojos.map(dojo =>
+          EventService.v3.get(dojo.dojoId, {
+            params: {
+              query,
+              related: 'sessions.tickets',
+            } }),
+        ));
+        this.events = events
+          .reduce((acc, dojoEvents) => acc.concat(dojoEvents.body.results), [])
+          .map(event => ({
+            ...event,
+            usersDojos: this.usersDojos.filter(usersDojo => usersDojo.dojoId === event.dojoId),
```

Comment:
Wouldn't that be easier to build an object with dojoId as a key ?
Or do it post-`slice` to avoid having to filter on every event ?

Reply:
Or both! I wasn't too concerned about looping over data given it shouldn't be _that_ much data, but any and all optimisations are always welcome!